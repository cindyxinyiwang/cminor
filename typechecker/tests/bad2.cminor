// expr type error

func: function integer() = {
	
}

a: function integer () = {
	x: integer = 1;
	
	y: string = "hello hi";
	a0: array[] integer = {1,2,3};
	b: array[3] array [2] integer = {{1,2}, {3,4}, {5,6}};
	z: char = 'b';
	p: boolean = true;
	q: boolean = false;

	// array
	c: array[3] array [2] array[1] integer = {{{true},{2}}, {{3},{4}}, {{5},{6}}};
	wrong: array[] boolean;
	wrong1: array[10] integer = {1,2};
	x = a0[z];
	
	// integer
	x1: integer = true;

	// assign
	a = func;

	// == !=
	c == b;
	a != func;
	a == func;
	c != b;

}