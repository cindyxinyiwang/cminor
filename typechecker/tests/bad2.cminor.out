x resovles to local 0
a0 resovles to local 2
z resovles to local 4
a resovles to global a
func resovles to global func
c resovles to local 7
b resovles to local 3
a resovles to global a
func resovles to global func
a resovles to global a
func resovles to global func
c resovles to local 7
b resovles to local 3
type error: array(a0) declaration must have a size value
type error: array of element type array [1] boolean cannot have expression of type array [1] integer
type error: array of element type array [2] array [1] boolean cannot have expression of type array [2] array [1] integer
type error: array of element type array [2] array [1] boolean cannot have expression of type array [2] array [1] integer
type error: cannot initialize variable (c) of type array [3] array [2] array [1] integer and assign it a array [3] array [2] array [1] boolean
type error: array(wrong) declaration must have a size value
type error: cannot use character as index to an array
type error: cannot initialize variable (x1) of type integer and assign it a boolean
type error: cannot assign function integer() to a function integer()
type error: cannot compare equality array [3] array [2] array [1] integer with a array [3] array [2] integer
type error: cannot compare non equality function integer() with a function integer()
type error: cannot compare equality function integer() with a function integer()
type error: cannot compare non equality array [3] array [2] array [1] integer with a array [3] array [2] integer
